axum-core/src/response/into_response.rs

```diff
@@ -396,6 +396,27 @@ where
     }
 }
 
+impl<R> IntoResponse for (http::response::Parts, R)
+where
+    R: IntoResponse,
+{
+    fn into_response(self) -> Response {
+        let (parts, res) = self;
+        (parts.status, parts.headers, res).into_response()
```

Comment:
Should we be replacing the version here? If not, why? The general pattern we try to follow is that things further to the left in the tuple takes precedence.

We did previously talk about `Version` and decided to just let hyper handle it or require the user to manually build a response. See https://github.com/tokio-rs/axum/issues/802.

I'm not sure what the right call is here.

Reply:
We could request that the http crate explicitly specifies that the default value is `HTTP_11` and add an assert for it 😄

Reply:
> I think we should probably also merge the headers with `HeaderMap::extend`.

`IntoResponseParts` does the right thing here. If there are any headers in the `HeaderMap`, then the headers from `R::into_response()` will be extended with those headers.

Reply:
Ah yes I see thats right!